### PR TITLE
Improve Pipelines SPA UX

### PR DESCRIPTION
**Pull request by kritika-singh3**
Issue: #7860 

Description:
 - Updated internal pipeline structures api v1 to return environment and dependant pipelines
 - Updated the Pipeline SPA to disable delete pipeline icon based on
      - origin:  if defined remotely
      - if a part of any environment
      - if has some pipelines dependant on it

### CODE REVIEW

**Comment by kritika-singh3** [comment]
Previews:

1. Delete icon when enabled 

delete-pipeline-2

2. Delete icon if pipeline is defined in a config-repo

delete-pipeline-1

3. Delete icon if pipeline is a part of environment

delete-pipeline-3

4. Delete icon when pipeline has downstream pipelines

delete-pipeline-4

File: api/api-internal-pipeline-structure-v2/build.gradle [hunk 1]
```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright 2019 ThoughtWorks, Inc.
```
Comment (GaneshSPatil) [inline api/api-internal-pipeline-structure-v2/build.gradle]: update copyright?

File: .gitignore [hunk 2]
```diff
@@ -521,3 +521,7 @@ api/api-packages-v2/out/
 api/api-packages-v2/target/
 api/api-packages-v2/logs/
 api/api-packages-v2/config/
+api/api-internal-pipeline-structure-v2/out/
```
Comment (GaneshSPatil) [inline .gitignore]: as we're introducing v2 of internal pipeline-structure API, should we remove `v1`?

Comment (kritika-singh3) [reply .gitignore]: Decided against bumping up the api. Updated the v1 itself to return the desired values.

**Comment by maheshp** [comment]
Verified on GoCD Version: 20.3.0 (11450-b09b089652fae2543cda69a9a788038547a19054).

**Pull request merged**